src/nvim/screen.c

```diff
@@ -956,16 +957,7 @@ void draw_tabline(void)
 
   // Use the 'tabline' option if it's set.
   if (*p_tal != NUL) {
-    int saved_did_emsg = did_emsg;
-
-    // Check for an error.  If there is one we would loop in redrawing the
-    // screen.  Avoid that by making 'tabline' empty.
-    did_emsg = false;
-    win_redr_custom(NULL, false, false);
-    if (did_emsg) {
-      set_string_option_direct("tabline", -1, "", OPT_FREE, SID_ERROR);
-    }
-    did_emsg |= saved_did_emsg;
```

Comment:
Why was this removed?

Reply:
The `did_emsg` check was moved to `build_stl_str_hl()` instead of around each `build_stl_str_hl()` call (by passing in the option name) https://github.com/neovim/neovim/pull/20621#discussion_r993498582.

Reply:
Sounds like a refactor that can be reviewed independently? 😉

Reply:
Opened https://github.com/vim/vim/pull/11467 to check with Bram.